cli/command/service/opts.go

```diff
@@ -660,6 +667,109 @@ func (options *serviceOptions) makeEnv() ([]string, error) {
 	return currentEnv, nil
 }
 
+func (options *serviceOptions) ToPrivileges(flags *pflag.FlagSet) (*swarm.Privileges, error) {
+	// we're going to go through several possible uses of the Privileges
+	// struct, which may or may not be used. If some stage uses it (after the
+	// first), we'll check if it's nil and create it if it hasn't been created
+	// yet.
+	var privileges *swarm.Privileges
+	if options.credentialSpec.String() != "" && options.credentialSpec.Value() != nil {
+		privileges = &swarm.Privileges{
+			CredentialSpec: options.credentialSpec.Value(),
+		}
+	}
+
+	if flags.Changed(flagNoNewPrivileges) {
+		if privileges == nil {
+			privileges = &swarm.Privileges{}
+		}
+		privileges.NoNewPrivileges = options.noNewPrivileges
+	}
+
+	if flags.Changed(flagAppArmor) {
+		if privileges == nil {
+			privileges = &swarm.Privileges{}
+		}
+		switch options.appArmor {
+		case "default":
+			privileges.AppArmor = &swarm.AppArmorOpts{
+				Mode: swarm.AppArmorModeDefault,
+			}
+		case "disabled":
+			privileges.AppArmor = &swarm.AppArmorOpts{
+				Mode: swarm.AppArmorModeDisabled,
+			}
+		default:
+			return nil, errors.Errorf(
+				"unknown AppArmor mode %q. Supported modes are %q and %q",
+				options.appArmor,
+				swarm.AppArmorModeDefault,
+				swarm.AppArmorModeDisabled,
+			)
+		}
+	}
+
+	if flags.Changed(flagSeccomp) {
+		if privileges == nil {
+			privileges = &swarm.Privileges{}
+		}
+		seccomp, err := options.ToSeccompOpts()
+		if err != nil {
+			return nil, err
+		}
+		privileges.Seccomp = seccomp
+	}
+
+	return privileges, nil
+}
+
+func (options *serviceOptions) ToSeccompOpts() (*swarm.SeccompOpts, error) {
+	switch arg := options.seccomp; arg {
+	case "default":
+		return &swarm.SeccompOpts{
+			Mode: swarm.SeccompModeDefault,
+		}, nil
+	case "unconfined":
+		return &swarm.SeccompOpts{
+			Mode: swarm.SeccompModeUnconfined,
+		}, nil
+	default:
+		dir, _ := filepath.Split(arg)
+		// if the directory is empty, this isn't a file path. Even though
+		// the user may be referring to a file in the local directory, for
+		// disambiguation's sake, we require a custom profile file to be
+		// given as a path.
+		if dir == "" {
+			// check if the file exists locally
+			if _, err := os.Stat(arg); errors.Is(err, os.ErrNotExist) {
+				return nil, errors.Errorf("unknown seccomp mode %q.", arg)
+			}
+			return nil, errors.Errorf(
+				"unknown seccomp mode %q. (did you mean custom a seccomp profile \"./%s\"?)",
+				arg, arg,
+			)
```

Comment:
I'm not sure why it is necessary to check if the file exists if we expect to error anyway.

I would just remove the if statement and return 
"unknown seccomp mode %q. a custom seccomp profile must be specified as a relative or absolute file path \"./profile.json\" or \"/path/profile.json\"."